src/Factory.sol

```diff
@@ -17,17 +17,20 @@ contract Factory is IFactory, Owned {
     /// @inheritdoc IFactory
     bool public canUpgrade = true;
 
-    /// @inheritdoc IFactory
-    address public implementation;
-
     /// @inheritdoc IFactory
     address public settings;
 
     /// @inheritdoc IFactory
     address public events;
 
     /// @inheritdoc IFactory
-    mapping(address accountOwner => address account) public ownerToAccount;
+    address public implementation;
+
+    /// @inheritdoc IFactory
+    mapping(address accounts => bool exist) public accounts;
+
+    /// @notice mapping of owner to accounts owned by owner
+    mapping(address owner => address[] accounts) private ownerAccounts;
```

Comment:
Do we have to keep this on onchain? Quite expensive

Reply:
I did not want to but FE really pushed for it